public/jsx/startup.jsx

```diff
@@ -10,19 +10,17 @@ var Startup = React.createClass({
   componentDidMount: function() {
     var self = this;
     require('electron').ipcRenderer.on('setup-status', function(evt, data) {
-        var s = self.state;
-        s.statusPython = data.python;
-        s.statusJupyter = data.jupyter;
+        var s = {
+          statusPython: data.python,
+          statusJupyter: data.jupyter
+        };
         if (data.python==false || data.jupyter==false) {
           s.status = "error";
         } else if (data.isFirstRun==true) {
           s.status = "good to go";
           setTimeout(function() {
             self.setState({
-              status: "tour",
-              statusPython: self.state.statusPython,
```

Comment:
The `setState` method performs a shallow merge into the previous state, so we can avoid enumerating all the fields of the state object.

https://facebook.github.io/react/docs/component-api.html#setstate